src/Microsoft.AspNetCore.Proxy/SharedProxyOptions.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+
+namespace Microsoft.AspNetCore.Proxy
+{
+    /// <summary>
+    /// Shared Proxy Options
+    /// </summary>
+    public class SharedProxyOptions
```

Comment:
> Why not just ProxyOptions?

I have taken this naming from AuthenticationMiddleware. Can go with ProxyOptions as well. What's your final verdict?